hibernate-core/src/main/java/org/hibernate/resource/beans/container/internal/CdiBeanContainerExtendedAccessImpl.java

```diff
@@ -41,8 +41,12 @@ private CdiBeanContainerExtendedAccessImpl(ExtendedBeanManager beanManager) {
 	protected <B> ContainedBeanImplementor<B> createBean(
 			Class<B> beanType,
 			BeanLifecycleStrategy lifecycleStrategy,
-			BeanInstanceProducer fallbackProducer) {
-		if ( usableBeanManager == null ) {
+			BeanInstanceProducer fallbackProducer,
+			boolean cdiRequiredIfAvailable) {
+		if ( !cdiRequiredIfAvailable ) {
```

Comment:
Shouldn't the `cdiRequiredIfAvailable` flag be ignored if `usableBeanManager` is non-null (and thus CDI is already available)?

E.g.:

```suggestion
		if ( !cdiRequiredIfAvailable && usableBeanManager == null ) {
```

That way if the caller is lucky and CDI was already initialized, they can get a CDI bean.